Software IRQs are missing from device.x leading to undefined symbols when linking

I believe that #8 missed adding the corresponding aliases for the `SW0_IRQ` - `SW5_IRQ` interrupt handlers into [device.x](https://github.com/rp-rs/rp235x-pac/blob/main/device.x) so causing undefined symbols unless the user of the HAL+PAC provides real/dummy interrupt handlers for the six software interrupts. 

I am experimenting with an RTIC application on an RP2350 board using the RP235x HAL & PAC. RTIC uses interrupt vectors for its "dispatchers" for software tasks (_i.e._ async fns that are spawned and not associated with real hardware interrupt handlers). Having got it working using hardware interrupts that I didn't need for their intended harware peripherals, it seemed sensible to use some of the software IRQs so I didn't constrain what hardware interrupts I can use in the future in my application. 

As soon as I started using the version of the rp235x-pac from GitHub that actually has the software IRQs defined, rather than V0.1.0 on crates.io, I immediately got undefined symbols for the `SWx_IRQ` symbols. From reading up on the [cortex_m_rt docs](https://docs.rs/cortex-m-rt/latest/cortex_m_rt/#devicex) this sounds like it's to be expected (unless I provide real or dummy interrupt handlers in my application).

I've patched a local copy of device.x to alias the relevant symbols and everything then works as expected.

Hope that's useful - if it would help I'm happy to create a PR...

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,7 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
-* Add SPAREIRQ_* interrupts
+* Add `SW*_IRQ` interrupts
 
 ## [0.1.0] 2024-08-18 ([Crates.io](https://crates.io/crates/rp235x-pac/0.1.0) | [Github](https://github.com/rp-rs/rp235x-pac/releases/tag/v0.1.0))
 

--- a/device.x
+++ b/device.x
@@ -42,4 +42,10 @@ PROVIDE(PLL_SYS_IRQ = DefaultHandler);
 PROVIDE(PLL_USB_IRQ = DefaultHandler);
 PROVIDE(POWMAN_IRQ_POW = DefaultHandler);
 PROVIDE(POWMAN_IRQ_TIMER = DefaultHandler);
+PROVIDE(SW0_IRQ = DefaultHandler);
+PROVIDE(SW1_IRQ = DefaultHandler);
+PROVIDE(SW2_IRQ = DefaultHandler);
+PROVIDE(SW3_IRQ = DefaultHandler);
+PROVIDE(SW4_IRQ = DefaultHandler);
+PROVIDE(SW5_IRQ = DefaultHandler);
 

--- a/sortFieldsAlphaNum.sh
+++ b/sortFieldsAlphaNum.sh
@@ -35,7 +35,9 @@ for ((i = 0; i < ${#alphaTargets[@]}; i++)); do
   # Calculate the tail number needed to crop to blockEnd
   blockTail=$((maxLen-blockEnd))
 
-  # This will replace the parts that need to be sorted.
+  # This will replace the parts that need to be sorted. Ensure that sort ordering is 
+  # consistent irrespective of the user's locale by using the "C" locale.
+  export LC_ALL=C
   toReplace=$(cat ${FILE} | grep "${alphaTargets[$i]}" -A $blockEndLine | tail -n $blockEndLine)
   if [ "$(uname)" == "Darwin" ]; then
     alphabetized=$(echo "$toReplace" | sed '$!N;s/\n/ /' | sort | sed 's/     /\n    /')

--- a/update.sh
+++ b/update.sh
@@ -33,6 +33,7 @@ rustfmt inner/lib.rs
 mv inner/lib.rs inner/mod_cortex_m.rs
 rm -rf ${SCRIPT_DIR}/src/inner
 mv inner ${SCRIPT_DIR}/src
+mv device.x ${SCRIPT_DIR}
 popd
 rm -rf ${tmp_dir}
 

